app/services/feeds/check_item_previously_imported.rb

```diff
@@ -0,0 +1,25 @@
+# Checks if an item has been previously imported for the given user
+# Item transformed as Article objects are unique per user, not globally
+module Feeds
+  class CheckItemPreviouslyImported
+    def self.call(item, user)
+      new(item, user).call
+    end
+
+    def initialize(item, user)
+      @item = item
+      @user = user
+    end
+
+    def call
+      title = item.title.strip.gsub('"', '\"')
+      feed_source_url = item.url.strip.split("?source=")[0]
```

Comment:
Are you sure? If the URL doesn't contains `?source` split will be the string:

```ruby
> "foobar".split("?source=")[0]
=> "foobar"
```

If the URL is blank yes this will be nil, but I think that's why it also looks for the title. We could make the query smarter to don't check both if one is missing but I think these optimizations can wait for later. I'll make a not for it!

BTW I just copied and pasted it from the existing code:

https://github.com/forem/forem/blob/cf619caf70f5ed129303b5b701b6e438913ed350/app/services/rss_reader.rb#L117-L122

Reply:
Yes, my mistake, I had only tested the blank case, not URLs which don't contain "?source=". I agree it can wait though.